### PR TITLE
Fix broken links on minikube website

### PR DESCRIPTION
**Pull request by prasadkatti**
Broken links - 

- https://minikube.sigs.k8s.io/reference/environment_variables/
- https://minikube.sigs.k8s.io/reference/networking/ (#7540)
- https://kubernetes.io/docs/Handbook/debug-application-cluster/audit/
- https://minikube.sigs.k8s.io/Handbook/sync/
- /docs/start/linux/#hypervisor-setup
- https://minikube.sigs.k8s.io/reference/configuration/kubernetes/
- /docs/contrib/roadmap/principles.md

I have fixed **some** of these links. For some links, I don't know what is the right new location.

### CODE REVIEW

File: site/content/en/docs/_index.md [hunk 1]
```diff
@@ -17,9 +17,9 @@ minikube quickly sets up a local Kubernetes cluster on macOS, Linux, and Windows
 * Cross-platform (Linux, macOS, Windows)
 * Deploy as a VM, a container, or on bare-metal
 * Multiple container runtimes (CRI-O, containerd, docker)
-* Docker API endpoint for blazing fast [image pushes](https://minikube.sigs.k8s.io/docs/handbook/pushing/#pushing-directly-to-the-in-cluster-docker-daemon)
-* Advanced features such as [LoadBalancer](https://minikube.sigs.k8s.io/Handbook/loadbalancer/), filesystem mounts, and FeatureGates
-* [Addons](https://minikube.sigs.k8s.io/Handbook/addons/) for easily installed Kubernetes applications
+* Docker API endpoint for blazing fast [image pushes]({{< ref "/docs/handbook/pushing.md#pushing-directly-to-the-in-cluster-docker-daemon" >}})
```
Comment (medyagh) [inline site/content/en/docs/_index.md:20]: thank you for using "ref" instead of FQDN

File: site/content/en/docs/handbook/addons.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+title: "Minikube Addons"
```
Comment (medyagh) [inline site/content/en/docs/handbook/addons.md:2]: minikube should always be lower case

File: site/content/en/docs/handbook/addons.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+weight: 7
+description: >
+  List of addons for minikube
+---
+
+List of addons can be found here - [list of addons](https://github.com/kubernetes/minikube/tree/master/deploy/addons)
```
Comment (medyagh) [inline site/content/en/docs/handbook/addons.md:8]: use document auto generate here (to automaticly list addons list from minikube)

Comment (prasadkatti) [reply site/content/en/docs/handbook/addons.md:8]: yes, I plan to tackle that next. Right now, I am having an issue where the addons page does not show up on the site. The link to the addons page yields a 404. All I did was I added a addons.md page in `docs/handbook`. Do I need to do anything else? Like `hugo new addons.md` or something?

Comment (medyagh) [reply site/content/en/docs/handbook/addons.md:8]: actually no need to use auto -generate here. never mind

Comment (medyagh) [reply site/content/en/docs/handbook/addons.md:8]: maybe that page was missing I see one here https://minikube.sigs.k8s.io/docs/handbook/deploying/

**Comment by prasadkatti** [comment]
@medyagh - this is ready to be reviewed/merged.

**Comment by prasadkatti** [comment]
@medyagh - added aliases

**Pull request merged**